os/kernel/debug/mem_leak_checker.c

```diff
@@ -52,46 +52,66 @@ struct alloc_node_info_s {
 	struct alloc_node_info_s *next;
 };
 
-static struct alloc_node_info_s *hash_table[HASH_SIZE];
-static struct alloc_node_info_s node_info[MAX_ALLOC_COUNT];
-static struct mem_leak_checker_info_s leak_checker;
-static int checkerpid;
+static struct alloc_node_info_s **g_hash_table;
+static struct alloc_node_info_s *g_node_info;
 
-#ifdef CONFIG_MM_KERNEL_HEAP
-extern struct mm_heap_s g_kmmheap[CONFIG_KMM_NHEAPS];
-#endif
-
-static void hash_init(void)
+static int hash_init(void)
 {
 	int index;
+
+	DEBUGASSERT(g_hash_table == NULL && g_node_info == NULL);
+
+	g_hash_table = (struct alloc_node_info_s **)malloc(sizeof(struct alloc_node_info_s *) * HASH_SIZE);
+	if (!g_hash_table) {
+		return ERROR;
+	}
+
 	for (index = 0; index < HASH_SIZE; ++index) {
-		hash_table[index] = NULL;
+		g_hash_table[index] = NULL;
 	}
```

Comment:
How about doing this after allocating the `g_node_info`? Because if the `g_node_info` has null, then no need to execute this.

Reply:
thank you! I changed